### ISSUE TITLE
top showed memory is much larger than heap profiler showed memory

### ISSUE THREAD

**Issue by legend0011** [facebook/rocksdb#5561]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://www.facebook.com/groups/rocksdb.dev

I use rocksdb in container, so the memory usage is really important for me. I found that rocksdb use much more memory than I expected. Therefore, I heap profile rocksdb and found that memory use in heap is 75MB, but pmap or top showed that it use 250MB+ memory, why?
How can I reduce the gap?

The gap between heap allocated memory and RES column in "top" command should be small. 

### Steps to reproduce the behavior
Just always Put() into rocksdb will reproduce the problem.
```
void WriteTestDB() {
  rocksdb::Options options;
  options.create_if_missing = true;
  rocksdb::DB* db = NULL;
  rocksdb::Status st = rocksdb::DB::Open(options, FLAGS_db_path, &db);
  CHECK(st.ok()) << "Fail to open leveldb:" << FLAGS_db_path;

  for (int i = 0; i < 1000000; ++i) {
    string key = StringPrintf("key_%03d", i);
    string value = StringPrintf("value_%03d", i);
    db->Put(rocksdb::WriteOptions(), key, value);
  }

  delete db;
}
```

**Comment by riversand963** [comment]
Does your heap profiler account for only memory allocated from the heap?
IIRC, pmap shows the memory map of the process which include libraries, etc. Can you share the command you used including the options?

**Comment by legend0011** [comment]
yes, I used google gperf tool, it only account for memory allocated from the heap.
pmap command I used is like this: pmap [pid]
the pmap result showed that a row labeled  [ anon ] whose memory is always growing, I don't know what it represents for, for there are other rows which are also labeled [ anon ], the row is like this:
```
00007fc148c5c000 205312K rw---   [ anon ]
```

**Comment by legend0011** [comment]
I know that pmap shows virtual memory usage, which is the same as column VIRT in 'top' result, so it will be much larger than we actually use. But the RES column in 'top' command is the actual memory we use, which is still much larger than heap profiler result.

And I understand that memory usage contains memory from stack and heap. The heap profiler shows memory allocated from heap. Therefore, my guess is,  the gap should be the memory allocated from stack.
 
Is there a chance that we reduce the memory allocated from stack ? I just don't understand why rocksdb use so much memory on stack and is always growing.
